options.go

```diff
@@ -232,12 +247,43 @@ func ParseURL(redisURL string) (*Options, error) {
 			return nil, fmt.Errorf("invalid redis database number: %q", f[0])
 		}
 	default:
-		return nil, errors.New("invalid redis URL path: " + u.Path)
+		return nil, fmt.Errorf("invalid redis URL path: %s", u.Path)
 	}
 
 	if u.Scheme == "rediss" {
 		o.TLSConfig = &tls.Config{ServerName: h}
 	}
+
+	return o, nil
+}
+
+func setupUnixConn(u *url.URL) (*Options, error) {
+	o := &Options{
+		Network: "unix",
+	}
+
+	if strings.TrimSpace(u.Path) == "" { // path is required with unix connection
+		return nil, errors.New("empty redis unix socket path")
+	}
+	o.Addr = u.Path
+
+	if u.User != nil {
```

Comment:
It looks like the rest of the func should be the same for `setupTCPConn` and `setupUnixConn`. Could you extract it to a separate func and use it from `setupTCPConn` and `setupUnixConn`?

Reply:
Not really. The same `user/password` part only.
But `Network` field we have to setup from u.Path because unix socket is a path to a file.
And we have to get database number different way: from u.Query.
I can move `user/password` logic to separate func but we'll have to write the same amount of code on call this func and it looks like redundant func from the first glance.
WDYT?

Reply:
It turned out that you were right.
I've moved `user/password` logic to separate func.
Take a look.
Thanks.